src/core/server/saved_objects/migrations/core/index_migrator.ts

```diff
@@ -207,6 +207,7 @@ async function migrateSourceToDest(context: Context) {
     await Index.write(
       client,
       dest.indexName,
+      // @ts-expect-error @opensearch-project/opensearch _source is optional
```

Comment:
If `_source` is optional why is there an error?

Reply:
```
Argument of type 'SearchHit<SavedObjectsRawDocSource>[]' is not assignable to parameter of type 'SavedObjectsRawDoc[]'.
  Type 'SearchHit<SavedObjectsRawDocSource>' is not assignable to type 'SavedObjectsRawDoc'.
    Types of property '_source' are incompatible.
      Type 'SavedObjectsRawDocSource | undefined' is not assignable to type 'SavedObjectsRawDocSource'.
        Type 'undefined' is not assignable to type 'SavedObjectsRawDocSource'.ts(2345)
```

Reply:
const docs = await read() and type is opensearchtypes.SearchHit<SavedObjectsRawDocSource>[]

SearchHit is defined in nodejs client as:
```
export interface SearchHit<TDocument = unknown> {
  _index: IndexName
  _id: Id
  _score?: double
  _type?: Type
  _explanation?: ExplainExplanation
  fields?: Record<string, any>
  highlight?: Record<string, string[]>
  inner_hits?: Record<string, SearchInnerHitsResult>
  matched_queries?: string[]
  _nested?: SearchNestedIdentity
  _ignored?: string[]
  _shard?: string
  _node?: string
  _routing?: string
  _source?: TDocument
  _seq_no?: SequenceNumber
  _primary_term?: long
  _version?: VersionNumber
  sort?: SearchSortResults
}
```
Here _source is optional. But migrateRawDocs function requires docs `rawDocs: SavedObjectsRawDoc[],`

src/core/server/saved_objects/serialization/types.ts :
```
export interface SavedObjectsRawDoc {
  _id: string;
  _source: SavedObjectsRawDocSource;
  _type?: string;
  _seq_no?: number;
  _primary_term?: number;
}
```
Here _source is not optional.

Reply:
Thanks for the explanation!